src/Tokenizer/Analyzer/Analysis/MethodAnalysis.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+/**
+ * @internal
+ */
+final class MethodAnalysis
+{
+    private ?int $visibility;
+    private bool $isStatic;
+    private bool $isAbstract;
+    private bool $isFinal;
+
+    public function __construct(?int $visibility, bool $isStatic, bool $isAbstract, bool $isFinal)
+    {
+        $this->visibility = $visibility;
+        $this->isStatic = $isStatic;
+        $this->isAbstract = $isAbstract;
+        $this->isFinal = $isFinal;
+    }
+
+    public function hasVisibility(): bool
```

Comment:
```suggestion
    public function hasExplicitVisibility(): bool
```
Visibility is always there, but may be explicit or implicit (public as default).

Reply:
> Visibility is always there

it depends - if the question is "what is the declared method's visibility?" or "what is the defined method's visibility?", the former is for static analysis and the latter for dynamic analysis and we are interested in the static analysis, aren't we?

Reply:
I don't get the difference. The signature is always static, you can't change it in runtime. So method either has explicit visibility level, or is implicitly public.

Reply:
I know, and my suggestion is exactly because of that 😅. As I said, technically visibility is always defined, so `hasVisibility()` is not descriptive enough (because technically it's always true).